src/Compilers/CSharp/Portable/BoundTree/BoundNullCoalescingOperatorResultKind.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal enum BoundNullCoalescingOperatorResultKind
```

Comment:
Perhaps add a comment: this is used to track the origin of the best type in null-coalescing operator, so it can be re-interpreted with nullability in mind.